fairmq/sdk/Topology.h

```diff
@@ -523,6 +706,7 @@ class BasicTopology : public AsioBase<Executor, Allocator>
         {
             if (!fOp.IsCompleted() && fCount == fTasks.size()) {
                 fTimer.cancel();
+                // TODO: this should be initiated differently
```

Comment:
?

Reply:
The `TryCompletion` method is called from `asio::async_initiate` directly and, if it can complete, it will complete the Op without directly. In the previous version we had special handling for thise case, at least for the StateChange Op. We discussed that we should also do it for the new Ops, but this is not yet implemented in this PR. I'll rephrase it to be more clear and move the comment to the place where TryCompletion is called.

Reply:
Updated/moved the comments inside the `async_initiate`.